src/inputs/netprobe/PingProbe.cpp

```diff
@@ -173,17 +186,25 @@ bool PingProbe::start(std::shared_ptr<uvw::Loop> io_loop)
             }
         });
 
-        _recv_connection = PingReceiver::recv_signal.connect([this](pcpp::Packet &packet, timespec stamp) {
-            _recv(packet, TestType::Ping, _name, stamp);
-        });
-
         (_sequence == UCHAR_MAX) ? _sequence = 0 : _sequence++;
         _send_icmp_v4(_internal_sequence);
         _internal_sequence++;
         _timeout_timer->start(uvw::TimerHandle::Time{_config.timeout_msec}, uvw::TimerHandle::Time{0});
         _internal_timer->start(uvw::TimerHandle::Time{_config.packets_interval_msec}, uvw::TimerHandle::Time{_config.packets_interval_msec});
     });
 
+    _recv_handler = _io_loop->resource<uvw::AsyncHandle>();
+    if (!_recv_handler) {
+        throw NetProbeException("PingProbe - unable to initialize AsyncHandle receiver");
+    }
+    _recv_handler->on<uvw::AsyncEvent>([this](const auto &, auto &) {
+        for (auto &[packet, stamp] : PingReceiver::recv_packets) {
+            _recv(packet, TestType::Ping, _name, stamp);
+        }
+    });
+    _receiver->register_async_callback(_recv_handler);
```

Comment:
so how often does/when this fire?

Reply:
I kept the 100ms timer on receiver thread... so it will be called every 100ms. But now on the right thread